test/test_ops.py

```diff
@@ -279,106 +280,198 @@ def _ref_test_helper(
+                    # Special-cases boolean comparisons
+                    if prims.utils.is_boolean_dtype(a.dtype):
+                        assert b.dtype is torch.bool
+                        return (a ^ b).sum()
 
-                same = (a == b)
-                if prims.utils.is_float_dtype(a.dtype) or prims.utils.is_complex_dtype(a.dtype):
-                    same = torch.logical_or(same, torch.logical_and(torch.isnan(a), torch.isnan(b)))
+                    same = (a == b)
+                    if prims.utils.is_float_dtype(a.dtype) or prims.utils.is_complex_dtype(a.dtype):
+                        same = torch.logical_or(same, torch.logical_and(torch.isnan(a), torch.isnan(b)))
 
-                actual_error = torch.where(same, 0, torch.abs(a - b)).sum()
-                return actual_error
+                    actual_error = torch.where(same, 0, torch.abs(a - b)).sum()
+                    return actual_error
 
-            ref_distance = 0
-            for a, b in zip(tree_flatten(ref_result)[0], tree_flatten(precise_result)[0]):
-                ref_distance = ref_distance + _distance(a, b)
+                ref_distance = 0
+                for a, b in zip(tree_flatten(ref_result)[0], tree_flatten(precise_result)[0]):
+                    ref_distance = ref_distance + _distance(a, b)
 
-            torch_distance = 0
-            for a, b in zip(tree_flatten(torch_result)[0], tree_flatten(precise_result)[0]):
-                torch_distance = torch_distance + _distance(a, b)
+                torch_distance = 0
+                for a, b in zip(tree_flatten(torch_result)[0], tree_flatten(precise_result)[0]):
+                    torch_distance = torch_distance + _distance(a, b)
 
-            # TODO: consider adding some tolerance to this comparison
-            msg = f"Reference result was farther ({ref_distance}) from the precise " \
-                  f"computation than the torch result was ({torch_distance})!"
-            self.assertTrue(ref_distance <= torch_distance, msg=msg)
+                # TODO: consider adding some tolerance to this comparison
+                msg = f"Reference result was farther ({ref_distance}) from the precise " \
+                      f"computation than the torch result was ({torch_distance})!"
+                self.assertTrue(ref_distance <= torch_distance, msg=msg)
 
         # Reports numerical accuracy discrepancies
         if ex is not None:
             msg = "Test passed because the reference was more accurate than the torch operator."
             warnings.warn(msg)
 
+    # Directly call each ref (in the context) and test it against the
+    # corresponding op (outside of the context). This test calls *different*
+    # entrypoints. Users are not expected to call the ref directly, this is only
+    # done to check for compatibility issues.
+    def _make_test_inputs_ref_op(self, op):  # default
+        # (ctx, no_ctx)
+        return ((op, op.torch_opinfo),)  # ref, op
+
```

Comment:
If this is the standard way to call things (as it's called within a different test), it may be worth it to make it the default value for `make_test_inputs`.